### ISSUE TITLE
Replace Category pages with search results pages

### ISSUE THREAD

**Issue by bobsilverberg** [mozilla/addons-frontend#8079]
This is a spin-off from #8050 which captures the decisions made and described at https://github.com/mozilla/addons-frontend/issues/8050#issuecomment-494960622.

We will be doing away with category pages which show multiple shelves for a given category. Instead, any link to a category page will result in a Search Results page being displayed, with the search criteria in the URL specified as being limited to the given category. 

These search results will be sorted by recommended extensions, followed by ADI.

~~When a user chooses a new sort criteria from the form on this search results page, we will ensure that the new search uses recommended extensions as the primary sort criteria in order to force all recommended add-ons to the top.~~ - This has been moved to #8084.

**Comment by bobsilverberg** [comment]
@jvillalobos I did not hide this change behind the same flag that we are using to hide all of the Recommended badge stuff. This seems like a change that we could land earlier, as it doesn't seem to depend on the implementation of all of the Recommended features. Please let me know if you agree, and if not, and you'd like to delay the implementation of this feature until all of the Recommended things ship, then I will add the flag logic to it.

**Comment by jvillalobos** [comment]
Yes, I agree this can land earlier.

**Comment by bobsilverberg** [comment]
@jvillalobos There is a shelf at the bottom of the home page which lists some theme categories:

![Screenshot 2019-05-28 10 51 51](https://user-images.githubusercontent.com/142755/58488015-a05a4b80-8136-11e9-892c-98582eba2514.png)

Each of these links to a Category landing page, which we were planning on doing away with, and replacing with a Search Results page. Can I also redirect each of these links in this shelf to a Search Results page, as opposed to a Category landing page? If we do that then we can do away with Category landing pages entirely.

**Comment by bobsilverberg** [comment]
Further to my comment above, I think that if we need to keep the Category page code, in order to support Category pages that are linked to via that shelf mentioned above, we could land changes to it after this patch lands. That is, if we want to update the Category page to replace the Featured shelf with a Recommended shelf (as opposed to just removing the Category page entirely), we could land that change once this patch lands. We can do that because the Category page will only ever be used for themes, and we already have recommended themes in prod because of the special collection.

It will still be simpler to just do away with the Category page entirely, and have this shelf direct to the Search Results page, but if we have to make the change at least we won't have to hide it behind a flag.

**Comment by jvillalobos** [comment]
Yes, let's make this change for theme category pages as well. The category URLs (like `/themes/nature/`) should still exist, however, and redirect to the right search page.

**Comment by bobsilverberg** [comment]
> Yes, let's make this change for theme category pages as well. The category URLs (like `/themes/nature/`) should still exist, however, and redirect to the right search page.

Ah, okay, that's not something I was planning, so thanks for pointing that out.

**Comment by bobsilverberg** [comment]
I'm going to open a separate issue for dealing with this shelf (as it's on the home page, not a landing page), and also to deal with this redirect.

**Comment by AlexandraMoga** [comment]
Verified fixed on -dev with Fx67, Win10x64 and Android 8.0

- extension and theme categories are opening now a search results page with results from the selected category
- the default sorting is `recommended,users` meaning that results will be displayed in the following format:
  -  first we have recommended add-ons, sorted by average daily users, followed by other add-ons from the selected category, also sorted by average daily users

![2019-05-31_13-02-29](https://user-images.githubusercontent.com/31961530/58698768-a0f81980-83a4-11e9-9796-cb555c107ded.gif)
